### PR TITLE
Adds contributors for plans

### PR DESCRIPTION
**Pull request by briri**
Addresses #2349.

These changes are dependent on the ROR/Fundref PR's new `identifiers` table

Changes proposed in this PR:
- Adds a `contributors` table (with f_key to `orgs`)
- Adds a `plans_contributors` join table that sets a bitflag `roles` field based on the CRediT Taxonomy (thanks for alerting us to that @xsrust)
- Adds the 3 new `ethical_issues` fields to the `plans` table
- Added factories/tests for the new models
- Rake upgrade task to convert all `data_contact` ('Data curation' role) and `principal_investigator` ('Investigation' role) information on a plan over to `contributors` and adds a 'Writing - original draft' role to all of the plan's authors.

Changes to model/UI forthcoming. Need to update the code that creates a plan or shares a plan to add these contributors.

### CODE REVIEW

File: app/models/plan.rb [hunk 1]
```diff
@@ -420,7 +427,7 @@ def owner
                   .administrator
                   .order(:created_at)
                   .pluck(:user_id).first
-    User.find(usr_id)
+    usr_id.present? ? User.find(usr_id) : nil
```
Comment (briri) [inline app/models/plan.rb]: this was an edge case error that came up with some test records (a plan with no owner)

**Comment by raycarrick-ed** [comment]
Looks good to me as far as I can tell.
My one (very minor) comment is that we use "firstname" in users but "given_name" here.
Not a big deal but a mild incnsistency.
Not merging so Sam and Marta can take a look.

**Comment by briri** [comment]
yes, good point @raycarrick-ed I'll change it to firstname

**Comment by briri** [comment]
rebased with workflow fixes

**Comment by xsrust** [comment]
Looks like everything is now passing the tests.

I have a question about the implimentation, It makes sense to me that a Plan has_many contributors, but it's unclear to me why a  contributor would `have_many` plans.  It seems like this would allow anybody who's editing a contributor to update all references to that contributor, which I believe would be unintended.

**Comment by briri** [comment]
Hmm. The original intention with using the plans_contributors join table was to keep the size of the contributors table down. I see your point though on a belongs_to relation being a better fit. I have switched it over in the api-v1 branch so closing this one out
